tests/TestCompiler.hs

```diff
@@ -53,29 +53,26 @@ import Text.Regex.Base
 import Text.Regex.TDFA (Regex)
 
 import TestUtils
-import Test.Tasty
-import Test.Tasty.Hspec
-import Test.Tasty.Golden (goldenVsString)
-
-main :: IO TestTree
-main = do
-  (supportModules, supportExterns, supportForeigns) <- setupSupportModules
-  passing <- passingTests supportModules supportExterns supportForeigns
-  warning <- warningTests supportModules supportExterns supportForeigns
-  failing <- failingTests supportModules supportExterns supportForeigns
-  return . testGroup "compiler" $ [passing, warning, failing]
+import Test.Hspec
+
+spec :: Spec
+spec = do
+  (supportModules, supportExterns, supportForeigns) <- runIO setupSupportModules
+
+  passingTests supportModules supportExterns supportForeigns
+  warningTests supportModules supportExterns supportForeigns
+  failingTests supportModules supportExterns supportForeigns
```

Comment:
In the future I would quite like to bundle up `supportModules`, `supportExterns`, and `supportForeigns` into a single record data type, as I imagine that would make all of this quite a bit neater.